pkg/apis/core/types_seed.go

```diff
@@ -109,11 +111,43 @@ type SeedBackup struct {
 	SecretRef corev1.SecretReference
 }
 
-// SeedDNS contains DNS-relevant information about this seed cluster.
+// SeedDNS contains the external domain and configuration for the DNS provider
 type SeedDNS struct {
 	// IngressDomain is the domain of the Seed cluster pointing to the ingress controller endpoint. It will be used
 	// to construct ingress URLs for system applications running in Shoot clusters.
-	IngressDomain string
+	// This will be removed in the next API version and replaced by spec.ingress.domain.
+	IngressDomain *string
+	// Provider configures a DNSProvider
+	Provider *SeedDNSProvider
+}
+
+// SeedDNSProvider configures a DNS provider
+type SeedDNSProvider struct {
+	// Type describes the type of the dns-provider, for example `aws-route53`
+	Type string
+	// SecretRef is a reference to a Secret object containing cloud provider credentials used for registering external domains.
+	SecretRef corev1.SecretReference
+	// Domains contains information about which domains shall be included/excluded for this provider.
+	Domains *DNSIncludeExclude
+	// Zones contains information about which hosted zones shall be included/excluded for this provider.
+	Zones *DNSIncludeExclude
+}
+
+// Ingress configures the Ingress specific settings of the Seed cluster
+type Ingress struct {
+	// Domain specifies the IngressDomain of the Seed cluster pointing to the ingress controller endpoint. It will be used
```

Comment:
```suggestion
	// Domain specifies the Ingress domain of the Seed cluster pointing to the ingress controller endpoint. It will be used
```
The word IngressDomain is from the SeedDNS section - maybe adjust the wording.